### PR TITLE
Feature/model-class-example

### PR DESCRIPTION
**Pull request by Pascal0902**
Example of model class which can be used with both tensorflow and pytorch

### CODE REVIEW

File: grandml/core/models/effnetv2.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from .. import register_model
+
+
+@register_model('resnet50')
```
Comment (sauravgarg540) [inline grandml/core/models/effnetv2.py:4]: resnet or efficientnet?

File: grandml/core/models/effnetv2.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+def efficientnetv2():
```
Comment (sauravgarg540) [inline grandml/core/models/effnetv2.py:5]: example looks good, but in future we should add out implementation of the code. Because if user wants he can anyways use pytorch or tensorflow implementation of the model in their code.

File: grandml/examples/train_example.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        # PyTorch loss functions like CrossEntropyLoss expect labels in LongTensors
+        y_train = torch.randint(0, n_classes, (n_samples,)).long()
+
+    else:
+        raise ValueError("Invalid framework")
+
+    # Get the model, loss, and optimizer from the registry
+    my_model = get_model(
+        name='resnet50',
+        loss='categorical_crossentropy',
```
Comment (sauravgarg540) [inline grandml/examples/train_example.py:32]: what if user has his own implementation of loss?

**Review by sauravgarg540 — COMMENTED** [review]
I think get model should able to give a possibility to not only call registered method but also a custom loss function.

File: grandml/core/model.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            assert (
+                self.torch is not None
+            ), "PyTorch is not installed but was specified as the framework."
+
+        self.framework = framework  # Now it's specified during initialization
+        self.model = model
+        self.loss = loss
+        self.optimizer = optimizer
+
+    def train(self, x, y):
```
Comment (sauravgarg540) [inline grandml/core/model.py:33]: I think it is better to have a Trainer class to train the model

**Review by sauravgarg540 — CHANGES_REQUESTED** [review]
